src/Core/Context/CurrentContext.cs

```diff
@@ -363,6 +369,19 @@ public bool ProviderUser(Guid providerId)
             return Providers?.Any(o => o.Id == providerId) ?? false;
         }
 
+        public async Task<Guid?> ProviderIdForOrg(Guid orgId)
+        {
+            if (Organizations.Any(org => org.Id == orgId))
+            {
+                return null;
+            }
+
+            var po = (await GetProviderOrganizations())
+                .FirstOrDefault(po => po.OrganizationId == orgId);
+
+            return po.ProviderId;
```

Comment:
Need a null check on `po`. I see a check against `Organization` but not `GetProviderOrganizations()`, so with `FirstOrDefault()` the result could be `null`.